### PR TITLE
initialized asyncio unix socket listening

### PR DESCRIPTION
**Pull request by TheJJ**
basic structure for asyncio messaging to s3tpc

### CODE REVIEW

File: s3tpd/config.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    def verify(self):
+        """
+        Verifies the validity of the loaded attributes
+        """
+
+        if (self.comm_socket_permissions and
+            not self.comm_socket_permissions.isnumeric()):
+            raise ValueError("socket permission must be an octal number")
+
+        return
```
Comment (andrekupka) [inline s3tpd/config.py:64]: this return is not necessary

File: s3tpd/toolmessage.py [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+        global MESSAGE_ID, MESSAGE_CLSID, MESSAGE_IDS
+
+        super().__init__(name, bases, classdict)
+        MESSAGE_IDS[MESSAGE_ID] = cls
+        MESSAGE_CLSID[name] = MESSAGE_ID
+        MESSAGE_ID += 1
+
+
+class Message(metaclass=MessageMeta):
+    """
```
Comment (fr3akout) [inline s3tpd/toolmessage.py]: Copy and paste code from kevinci?

File: s3tpd/toolprotocol.py [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+    def log(self, msg, level=logging.INFO):
+        """ logs something for this connection """
+        logging.log(level, "[\x1b[1m%3d\x1b[m] %s" % (self.conn_id, msg))
+
+    async def process_messages(self):
+        """
+        Fetch messages from the queue and react to their request.
+        """
+
+        while not self.disconnected.done():
```
Comment (andrekupka) [inline s3tpd/toolprotocol.py:92]: This method is quite long and deeply nested. Maybe extract some methods

**Comment by TheJJ** [comment]
will be done in c++, python is dropped.

**Pull request closed without merging**